### PR TITLE
Revise attributions of Carto, Stamen and Open Street Map styles

### LINKED ISSUE #5675: OpennStreetMap credit
Hi,

With the introduction of 09343ae841d33c8e5a6c29445730af5db2f19aeb , Plotly maps can use the cartodb-dark style (now carto-darkmatter), and they get credit with "© CARTO", linking to https://carto.com/ .

I am under the impression that attribution is wrong. If I understand correctly, the maps used are stilll OSM's, and to fulfill OSM's credit criteria, credits shoud still be given to the OSM contributors: https://www.openstreetmap.org/copyright .

If I'm reading it right, CARTO seems to agree, as you can see in their own examples: https://carto.com/help/working-with-data/attribution/ .

This should be easilly fixed, with something like:
```
-                attribution: '<a href="https://carto.com/" target="_blank">© CARTO</a>',
+                attribution: '&copy; <a href="https://www.openstreetmap.org/copyright" target="_blank">OpenStreetMap</a> contributors, <a href="https://carto.com/" target="_blank">© CARTO</a>',
```

### PR DESCRIPTION
**Pull request by archmoj**
Fixes #5675.

@plotly/plotly_js

### CODE REVIEW

**Comment by alexcjohnson** [comment]
AFAICT only the Carto attributions should change, the Stamen ones were already using the correct attributions as described in http://maps.stamen.com/. They're a little long, but feels to me as though we should stick to their desired attributions.

**Comment by nicolaskruchten** [comment]
For Carto, just add whatever OSM requires :)

File: test/jasmine/tests/mapbox_test.js [hunk 1]
```diff
@@ -1487,6 +1548,11 @@ describe('mapbox plots', function() {
                 var s = d3SelectAll('.mapboxgl-ctrl-attrib');
                 expect(s.size()).toBe(1);
                 expect(s.text()).toEqual('© Mapbox © OpenStreetMap Improve this map');
+                assertLinks(s, [
+                    'https://www.mapbox.com/about/maps/',
+                    'http://www.openstreetmap.org/about/',
```
Comment (archmoj) [inline test/jasmine/tests/mapbox_test.js:1553]: @alexcjohnson any suggestion on how to replace this `http` link with `https`?

Comment (nicolaskruchten) [reply test/jasmine/tests/mapbox_test.js:1553]: if this is coming from MapboxGL.js let's just leave it for now

Comment (archmoj) [reply test/jasmine/tests/mapbox_test.js:1553]: Yeah. It's coming from there.

**Review by alexcjohnson — APPROVED** [review]
💃

**Pull request merged**